src/stan/services/optimize/laplace_sample.hpp

```diff
@@ -121,6 +121,8 @@ void laplace_sample(const Model& model, const Eigen::VectorXd& theta_hat,
 
     double log_p;
     if (calculate_lp) {
+      // clean up created vars after scope exit
+      stan::math::nested_rev_autodiff stack;
       log_p = log_density_fun(unc_draw).val();
```

Comment:
Since the actual error comes from the implicit cast in `log_density_fun`'s signature can we instead change `log_density_fun` to have an `auto&` argument and do an explicit promotion here from double to var?

```c++
  auto log_density_fun
      = [&](auto& theta) {
          return model.template log_prob<true, jacobian, stan::math::var>(
              theta,
              &log_density_msgs);
        };
```


```suggestion
      // clean up created vars after scope exit
      stan::math::nested_rev_autodiff stack;
      log_p = log_density_fun(unc_draw.cast<stan::math::var>()).val();
```